packages/lisk-p2p/src/peer_pool.ts

```diff
@@ -253,33 +256,74 @@ export class PeerPool extends EventEmitter {
 		});
 	}
 
+	public async fetchStatusAndCreatePeer(
+		seedPeers: ReadonlyArray<P2PPeerInfo>,
+		nodeInfo: P2PNodeInfo,
+		peerConfig: PeerConfig,
+	): Promise<ReadonlyArray<P2PDiscoveredPeerInfo>> {
+		const seedPeerUpdatedInfosAndSocket = await Promise.all(
+			seedPeers.map(async seedPeer =>
+				connectAndFetchPeerInfo(seedPeer, nodeInfo, peerConfig),
```

Comment:
Because `connectAndFetchPeerInfo` is a pass or fail operation, it should throw; that way the function signature doesn't need to return dual types and we don't need to use `instanceof` statement to separate the two outputs.

The `fetchStatusAndCreatePeer` function can succeed even if it has errors, so returning the error should happen at this level. Maybe this function could return an object with an `errors` and a `peerInfoList` field. If no errors occurred, then the `errors` field would be an empty array. Also, it doesn't seem that we are using the errors from `connectAndFetchPeerInfo` anywhere at the moment (and this function is not returning them).

We should emit them on the top level `P2P` instance using the existing `EVENT_FAILED_TO_FETCH_PEER_INFO` event.